.travis.yml

```diff
@@ -5,12 +19,17 @@ rvm:
 - 2.0.0
 
 cache:
-  directories:
-  - $HOME/gcloud/
-  - vendor/bundle
+  - directories:
+    - $HOME/gcloud/
+    - vendor/bundle
 
 env:
-- PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json #Other environment variables on same line
+- TEST_DIR=storage
+  PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
+  GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
+- TEST_DIR=pubsub
+  PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
+  GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
```

Comment:
While I can't say I'm a huge fan (it'll complicate the build UI by adding lots of rows), I'm not hugely against it. However, you should use the `global` keyword for env:

```yaml
env:
  global:
    - PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
    - GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
  matrix:
    - TEST_DIR=storage
    - TEST_DIR=pubsub
```